app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoMediaPlayerGlue.java

```diff
@@ -254,6 +383,37 @@ public void releaseResources() {
         }
     }
 
+    public void createMediaSessionIfNeeded() {
+        if (mVideoSession == null) {
+            mVideoSession = new MediaSession(this.getContext(), "VideoPlayer Session");
```

Comment:
Shouldn't  if (mVideoSession == null) check handle this case? A new media session is only created if it's already being released. Releasing a MediaSession happens in two place: fragment's onStop() and surfaceDestroyed().

The call to  createMediaSessionIfNeeded happens from two places: fragment's onStart() and prepareIfNeededAndPlay() (which is before playing a media item).  The latter is needed since it's possible that surfaceDestoryed is called without onStop() being called (If I'm playing a video in gregarious mode and press play/pause button and immediately click on NowPlayingCard for that video, surfaceDestroyed is called but before the fragment has a chance to stop I've returned to the activity and no onStart() call happens).